src/options.c

```diff
@@ -37,9 +38,10 @@ int opt_meth_setoption(lua_State *L, p_opt opt, p_socket ps)
     while (opt->name && strcmp(name, opt->name))
         opt++;
     if (!opt->func) {
-        char msg[45];
+        char* msg = malloc(30+strlen(name));
         sprintf(msg, "unsupported option `%.35s'", name);
         luaL_argerror(L, 2, msg);
+        free(msg);
```

Comment:
This is memory leak. Use push string, free and then error.

Reply:
`lua_pushfstring()` would be a leak-free (and more convenient) alternative. Or just make the buffer large enough -- at least 56 characters if I counted correctly.